### PR TITLE
Refactor ALL code

### PR DESCRIPTION
**Pull request by LumpBloom7**
This PR moves systems, components, and graphics into their respective namespaces.

Some preprocessor stuff has also been cleaned up. 

Also fix some retarded mistakes along with cleaning stray spaces.

Upd. by @xapdkop:
Also let's use fuckin' google c++ code guidelines.

### CODE REVIEW

**Review by kasmacioma — DISMISSED** [review]
1. Is `#include "stdIncludes.h"` needed into Components?
2. What is it?https://github.com/BloomCreativeWorks/BloomFramework/blob/ba5d6ba029e1d055fb9dd48fd4ce70391941055f/Test%20Bench/GameObjectTest/RandomizerSystem.h#L8
https://github.com/BloomCreativeWorks/BloomFramework/blob/ba5d6ba029e1d055fb9dd48fd4ce70391941055f/Test%20Bench/GameObjectTest/RandomizerSystem.h#L9
https://github.com/BloomCreativeWorks/BloomFramework/blob/ba5d6ba029e1d055fb9dd48fd4ce70391941055f/Test%20Bench/GameObjectTest/RandomizerSystem.h#L10

**Comment by kasmacioma** [comment]
And it's possible to remove `graphics::` from https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/include/Game.h#L11
because of this
https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/include/Game.h#L7

**Comment by LumpBloom7** [comment]
Noted.

**Comment by kasmacioma** [comment]
And what about spaces at the end of https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/include/stdIncludes.h#L5
and
https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/include/stdIncludes.h#L7
?

**Comment by kasmacioma** [comment]
Ahahahaha, why we don't fix it yet?
https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/src/Graphics/Texture.cpp#L13
https://github.com/BloomCreativeWorks/BloomFramework/blob/91b8696731237a54baff5bd2fcdb4b684ff0ac0b/BloomFramework/src/Graphics/Texture.cpp#L15

**Comment by kasmacioma** [comment]
Also, is this https://github.com/BloomCreativeWorks/BloomFramework/blob/1769489d4ac6d9f70f4cd53ee38df10289ab8bfb/BloomFramework/include/Graphics/Texture.h#L13
a good idea? As this function can be called by the user and can (or even should) break the program.

Upd. Ok, it doesn't break the program. But still this func should be removed.

**Comment by kasmacioma** [comment]
Omfg,
Codefactor:
> Don't use blank lines when you don't have to.

Google C++ Style Guide:
> Minimize use of vertical whitespace.
The basic principle is: The more code that fits on one screen, the easier it is to follow and understand the control flow of the program. Use whitespace purposefully to provide separation in that flow.

But when i try to remove blank line, codefactor mark it as issue.

**Comment by LumpBloom7** [comment]
@xapdkop Well there are exceptions.
Vertical blank space can be used if it improves readability or separates something. A blank line separating the public and private stuff makes sense.

And CodeFactor actually checks based on Google Style Guidelines.

**Comment by kasmacioma** [comment]
@LumpBloom7 hmmm, https://google.github.io/styleguide/cppguide.html#Class_Format
>Except for the first instance, these keywords should be preceded by a blank line. This rule is optional in small classes.

And if u want to follow google's guideline, u need to **completely rework** code.

**Comment by LumpBloom7** [comment]
We should already be following that rule no?

**Comment by kasmacioma** [comment]
> We should already be following that rule no?

Hmmm, following this? Ok, wait. We'll be following these rules.

**Comment by LumpBloom7** [comment]
Yup. We'll be following these to the best of our ability. 
But for the most part, CodeFactor is doing its job and our code and our code should be decent already.

**Comment by kasmacioma** [comment]
@LumpBloom7 ohh, it seems that we need to remove all c++17 features (but not now) [according to google's guidelines](https://google.github.io/styleguide/cppguide.html#C++_Version)
>Currently, code should target C++11, i.e., should not use C++14 or C++17 features

**Review by kasmacioma — APPROVED** [review]
just fuck google

**Comment by LumpBloom7** [comment]
@xapdkop Yeah... Not happening. 

We're following the style and formatting, but we're definitely not gonna sacrifice C++17 functionality for that.  

Also if the guidelines tells us not to use a certain function just because most of their legacy code doesn't make use of it, just ignore it. They shouldn't tell us to not use something because they can't.

TLDR follow formatting and naming etc. Nothing else.
